recipes/7zip/19.00/conanfile.py

```diff
@@ -32,28 +32,29 @@ def source(self):
     def build_requirements(self):
         self.build_requires("lzma_sdk/9.20")
 
-        if self.settings.compiler !=  "Visual Studio" and tools.os_info.is_windows and "make" not in os.environ.get("CONAN_MAKE_PROGRAM", ""):
+        if self.settings.compiler != "Visual Studio" and tools.os_info.is_windows and "make" not in os.environ.get("CONAN_MAKE_PROGRAM", ""):
             self.build_requires("make/4.2.1")
 
     def _uncompress_7z(self, filename):
         self.run("7zr x {}".format(filename))
 
-    _msvc_platforms = {
-        "x86_64": "x64",
-        "x86": "x86",
-    }
+    @property
+    def _msvc_platform(self):
+        return {
+            'x86_64': 'x64',
+            'x86': 'x86',
+        }[str(self.settings.arch)]
```

Comment:
```suggestion
        }.get(str(self.settings.arch))
```

Avoid index error when using other archs

Reply:
You avoid an IndexError, but you get a None instead.
Impossible architectures should be filtered out in `config_options` and `configure` anyways.

Reply:
I agree with @madebr that this should already be guarded against by `configure`. I can change it anyways if you strongly prefer the `get`-style, but it shouldn't really matter.

Reply:
So add an except treatment for IndexError, like ConanInvalidConfiguration when not supported yet.